aten/src/ATen/native/quantized/cpu/qconv.cpp

```diff
@@ -574,22 +560,49 @@ class QConvInt8 final {
       auto bias_fp32 = pack_data.bias;
       int8_t* w_data =
           reinterpret_cast<int8_t*>(weight_contig.data_ptr<c10::qint8>());
+
+      std::tie(pack_data.w_zero_points, pack_data.w_scales) =
+          make_zero_points_and_scales_tensor(weight_contig);
+      uint8_t* weight_zp_data =
+        (uint8_t*)pack_data.w_zero_points.data_ptr<c10::quint8>();
+      float* weight_scales_data = pack_data.w_scales.data_ptr<float>();
+      pack_data.requantization_scale =
+          generate_requantization_scales(
+              pack_data.w_scales, input_scale, output_scale);
+
       Tensor qnnp_weight = at::_empty_affine_quantized(
           weight_contig.sizes(),
           at::device(kCPU)
              .dtype(kQUInt8)
              .memory_format(MemoryFormat::ChannelsLast),
-          kernel_scale,
-          kernel_zp,
+          weight_scales_data[0],
+          weight_zp_data[0],
           c10::nullopt);
       auto* qnnp_w_data = qnnp_weight.data_ptr<c10::quint8>();
       auto wt_numel = weight_contig.numel();
       for (int i = 0; i < wt_numel; ++i) {
         qnnp_w_data[i] = static_cast<c10::quint8>(w_data[i] + 128);
       }
+
       // Original bias was float, so we requantize it here.
       auto bias = at::quantize_per_tensor(
-          bias_fp32, kernel_scale * input_scale, 0, kQInt32);
+          bias_fp32,
+          weight_scales_data[0] * input_scale,
+          0, kQInt32);
+
+      qnnpack::conv_param_t conv_p(
```

Comment:
Where is this used?

Reply:
In PrePackConvWeights.

Reply:
We create another instance of the param struct to pass to the run function. Can we avoid that?

Reply:
Yes. Let me see why it is done twice. I dont quite recall now, but will try to get rid of one if it is not there for a reason, else will add a comment.